cmd/utils/nodecmd/util.go

```diff
@@ -0,0 +1,220 @@
+package nodecmd
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/klaytn/klaytn/accounts/keystore"
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/cmd/utils"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/common/hexutil"
+	"github.com/klaytn/klaytn/consensus/istanbul"
+	"github.com/klaytn/klaytn/crypto"
+	"github.com/klaytn/klaytn/crypto/sha3"
+	"github.com/klaytn/klaytn/governance"
+	"github.com/klaytn/klaytn/params"
+	"github.com/klaytn/klaytn/rlp"
+	"gopkg.in/urfave/cli.v1"
+)
+
+const pgmInput = `
+<extra> <header file (json format)>
+<vote>  <bytes>
+<gov>   <bytes>
+<key>   <keystore path> <password>
+`
+
+var ErrInvalidParam = errors.New("Invalid length of parameter")
+
+var UtilCommand = cli.Command{
+	Action:    utils.MigrateFlags(parse),
+	Name:      "util",
+	Usage:     "offline utility " + pgmInput,
+	ArgsUsage: " ",
+	Category:  "MISCELLANEOUS COMMANDS",
+}
+
+func hex2Bytes(s string) []byte {
+	if data, err := hexutil.Decode(s); err == nil {
+		return data
+	} else {
+		panic(err)
+	}
+}
+
+func printUsage() error {
+	fmt.Println("Usage: ./kxn util " + pgmInput)
+	return ErrInvalidParam
+}
+
+func validateInput(ctx *cli.Context, parseTyp string) error {
+	switch parseTyp {
+	case "vote", "extra", "gov":
+		if len(ctx.Args()) != 2 {
+			return printUsage()
+		}
+	case "key":
+		if len(ctx.Args()) != 3 {
+			return printUsage()
+		}
+	}
+	return nil
+}
+
+func parse(ctx *cli.Context) error {
+	parseTyp, m := ctx.Args().Get(0), make(map[string]interface{})
+	if err := validateInput(ctx, parseTyp); err != nil {
+		return err
+	}
+	switch parseTyp {
+	case "vote":
+		data := ctx.Args().Get(1)
+		if err := parseVote(m, hex2Bytes(data)); err != nil {
+			return err
+		}
+	case "extra":
+		headerFile := ctx.Args().Get(1)
+		if err := parseExtra(m, headerFile); err != nil {
+			return err
+		}
+	case "gov":
+		data := ctx.Args().Get(1)
+		if err := parseGov(m, hex2Bytes(data)); err != nil {
+			return err
+		}
+	case "key":
+		keystorePath, passwd := ctx.Args().Get(1), ctx.Args().Get(2)
+		if err := extractKeypair(m, keystorePath, passwd); err != nil {
+			return err
+		}
+	default:
+		return printUsage()
+	}
+	pp(m)
+	return nil
+}
+
+func pp(m map[string]interface{}) {
```

Comment:
Does `pp` stand for pretty-print? How about changing its name to `prettyPrint`?

Reply:
Done.